### PR TITLE
fix: Improve Checkbox and Select behaviors

### PR DESCRIPTION
**Pull request by przemvs**
## Pull Request Checklist

- [ ] My code is covered by tests
- [ ] I will [merge the PR as breaking change](https://github.com/wireapp/wire-web-packages/wiki/Releases#create-major-release), if the API contract changes

### CODE REVIEW

File: packages/react-ui-kit/src/Form/Checkbox.tsx [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 /** @jsx jsx */
 import {jsx} from '@emotion/react';
-import React, {useId} from 'react';
+import React from 'react';
```
Comment (atomrc) [inline packages/react-ui-kit/src/Form/Checkbox.tsx:22]: Why removing the `useId`? It seems better than `Math.random().toString()`

File: packages/react-ui-kit/src/Form/Select.tsx [hunk 2]
```diff
@@ -27,23 +27,25 @@ import {inputStyle} from './Input';
...
   value: string | number;
   label: string;
+  description?: string;
 };
 
-export interface SelectProps {
+export interface SelectProps<T extends SelectOption> {
   id: string;
-  onChange: (selectedOption: string | number) => void;
+  onChange: (selectedOption: T['value']) => void;
```
Comment (atomrc) [inline packages/react-ui-kit/src/Form/Select.tsx:38]: :+1:

File: packages/react-ui-kit/src/Form/Select.tsx [hunk 3]
```diff
@@ -135,7 +137,7 @@ const filterSelectProps = props => filterProps(props, ['markInvalid']);
 
 const placeholderText = '- Please select -';
 
-export const Select = ({
+export const Select = <T extends SelectOption>({
```
Comment (atomrc) [inline packages/react-ui-kit/src/Form/Select.tsx:140]:
```suggestion
export const Select = <T extends SelectOption = SelectOptions>({
```

File: packages/react-ui-kit/src/Form/Select.tsx [hunk 4]
```diff
@@ -242,13 +247,22 @@ export const Select = ({
     };
   }, []);
 
+  useEffect(() => {
+    if (value) {
+      const valueIdx = options.findIndex(option => option.value === value.value);
+      setSelectedOption(valueIdx);
+    }
+  }, [options, value]);
```
Comment (atomrc) [inline packages/react-ui-kit/src/Form/Select.tsx:255]: You don't need a `useEffect` here. You are just creating new state from existing state, there is no effect there

**Review by atomrc — CHANGES_REQUESTED** [review]

File: packages/react-ui-kit/src/Form/Checkbox.tsx (hunk 1 above)
Comment (atomrc) [reply packages/react-ui-kit/src/Form/Checkbox.tsx:22]: What is the error exactly?

**Review by atomrc — APPROVED** [review]

File: packages/react-ui-kit/src/Form/Checkbox.tsx (hunk 1 above)
Comment (tlebon) [reply packages/react-ui-kit/src/Form/Checkbox.tsx:22]: did this get resolved?

File: packages/react-ui-kit/src/Form/Checkbox.tsx [hunk 5]
```diff
@@ -93,10 +93,11 @@ interface CheckboxProps<T = HTMLInputElement> extends InputProps<T> {
 
 const filterCheckboxProps = (props: CheckboxProps) => filterProps(props, ['markInvalid']);
 
+// We use Math.random..., because some of apps doesn't migrated to newest version of React.
```
Comment (tlebon) [inline packages/react-ui-kit/src/Form/Checkbox.tsx:96]: resolves comment on L22

**Pull request merged**